internal/repository/repository.go

```diff
@@ -0,0 +1,10 @@
+package internal
```

Comment:
The package here, instead of using `internal` for everything, a good practice would be to use the same as the folder where this code is sitting by.

Something like:
```suggestion
package repository
```

I guess if you make this change itself, all the code will fall apart. Can you fix the further issues this will cause?